src/v/cloud_roles/refresh_credentials.h

```diff
@@ -31,6 +31,11 @@ struct retry_params {
 
 class refresh_credentials {
 public:
+    enum class client_tls_enabled : bool {
+        yes = true,
+        no = false,
+    };
```

Comment:
@abhijat this looks like a case for using `seastar::bool_class<..>`

Reply:
yeah that looks like a better fit. created ticket https://github.com/redpanda-data/redpanda/issues/6384 for the change